render/egl.c

```diff
@@ -358,6 +358,58 @@ struct wlr_egl *wlr_egl_create(EGLenum platform, void *remote_display,
 	return NULL;
 }
 
+struct wlr_egl *wlr_egl_create_from_drm_fd(int drm_fd) {
+	PFNEGLQUERYDEVICESTRINGEXTPROC eglQueryDeviceStringEXT;
+	PFNEGLQUERYDEVICESEXTPROC eglQueryDevicesEXT;
+
+	load_egl_proc(&eglQueryDevicesEXT, "eglQueryDevicesEXT");
+	load_egl_proc(&eglQueryDeviceStringEXT,"eglQueryDeviceStringEXT");
+
+	static const EGLint MAX_DEVICES = 64;
+	EGLint nb_devices = 0;
+
+	eglQueryDevicesEXT(MAX_DEVICES, NULL, &nb_devices);
+
+	wlr_log(WLR_INFO, "nb_devices %d", nb_devices);
+
+	EGLDeviceEXT *devices = calloc(nb_devices, sizeof(EGLDeviceEXT));
+	if (devices == NULL) {
+		wlr_log_errno(WLR_ERROR, "Failed to allocate device list");
+		return NULL;
+	}
+
+	eglQueryDevicesEXT(MAX_DEVICES, devices, &nb_devices);
+
+	drmDevicePtr device = NULL;
+	drmGetDevice(drm_fd, &device);
+	wlr_log(WLR_INFO, "drmdeviceptr size %d", device->available_nodes);
+
+	for (int j = 0; j < DRM_NODE_MAX; j++)
+		if (device->available_nodes & 1 << j)
+			wlr_log(WLR_INFO, "nodes[%d] %s", j, device->nodes[j]);
+
+	EGLDeviceEXT *egl_device = NULL;
+	for(int i = 0; i < nb_devices; i++)
+	{
+		const char *dev_name = eglQueryDeviceStringEXT(devices[i],
+				EGL_DRM_DEVICE_FILE_EXT);
+
+		// TODO is nodes[0] always the card name?
+		if (strcmp(dev_name, device->nodes[0]) == 0) {
```

Comment:
I think there's a function called `device_has_name` (also in egl.c) which performs this check for you

(incidentally I also had a go at making this change before I saw your PR - maybe there's something of use here: https://github.com/swaywm/wlroots/compare/master...neon64:egl-from-drm-fd

Reply:
thanks for the tip!